docs/painless/painless-contexts/painless-runtime-fields-context.asciidoc

```diff
@@ -0,0 +1,218 @@
+{"theatre":"Down Port","play":"Driving","actors":["James Holland","Krissy Smith","Joe Muir","Ryan Earns"],"date":"2018-4-1","time":"3:00PM","row":1,"number":10,"cost":30,"sold":false}
+{"create":{"_index":"seats","_id":"11"}}
+{"theatre":"Down Port","play":"Driving","actors":["James Holland","Krissy Smith","Joe Muir","Ryan Earns"],"date":"2018-4-1","time":"3:00PM","row":1,"number":11,"cost":30,"sold":false}
+{"create":{"_index":"seats","_id":"12"}}
+----
+// TEST[continued]
+
+[[painless-runtime-fields-definition]]
+==== Define a runtime field with a Painless script
+Run the following request to define a runtime field named `day_of_week`. This
+field contains a script with the same `source` defined in
+<<painless-field-context,Field context>>, but also uses an `emit` function
+that runtime fields require when defining a Painless script.
+
+Because `day_of_week` is a runtime field, it isn't indexed, and the included
+script only runs at query time:
+
+[source,console]
+----
+PUT seats/_mapping
+{
+    "dynamic": "runtime",
+    "runtime": {
+      "day_of_week": {
+        "type": "date",
+        "script": {
+          "source": "emit(doc['datetime'].value.getDayOfWeekEnum().getValue())"
+        }
+      }
+    }
+}
+----
+// TEST[continued]
+
+After defining the runtime field and script in the mappings, you can run a
+query that includes `day_of_week`. When the query runs, {es} evaluates the
+included Painless script and dynamically generates a value based on the script
+definition:
+
+[source,console]
+----
+GET seats/_search
+{
+  "size": 2,
+  "fields": [
+    "time",
+    "day_of_week"
+  ],
+  "_source": false
+}
+----
+// TEST[continued]
+
+The response includes `day_of_week` for each hit. {es} calculates the value for
+this field dynamically at search time by operating on the `datetime` field
+defined in the mappings.
+
+[source,console-result]
+----
+{
+  ...
+  "hits" : {
+    "total" : {
+      "value" : 11,
+      "relation" : "eq"
+    },
+    "max_score" : 1.0,
+    "hits" : [
+      {
+        "_index" : "seats",
+        "_id" : "1",
+        "_score" : 1.0,
+        "fields" : {
+          "time" : [
+            "3:00PM"
+          ],
+          "day_of_week" : [
+            "1970-01-01T00:00:00.007Z"
+          ]
+        }
+      },
+      {
+        "_index" : "seats",
+        "_id" : "2",
+        "_score" : 1.0,
+        "fields" : {
+          "time" : [
+            "3:00PM"
+          ],
+          "day_of_week" : [
+            "1970-01-01T00:00:00.007Z"
+          ]
+        }
+      }
+    ]
+  }
+}
+----
+// TESTRESPONSE[s/\.\.\./"took" : $body.took,"timed_out" : $body.timed_out,"_shards" : $body._shards,/]
+// TESTRESPONSE[s/"value" : "11"/"value": $body.hits.total.value/]
+
+This is just one example of how you can use runtime fields. You can include
+Painless scripts in the context of runtime fields to override values for
+indexed fields or create fields that exist only as part of the query.
```

Comment:
I wonder if these details are needed here as we already link to the runtime fields docs.

Reply:
I'm ok with removing this bit.